src/main/java/org/jetbrains/appenders/JsonLayout.java

```diff
@@ -33,137 +33,124 @@
+        }
 
-        LocationField(String val) {
-            this.val = val;
+        void enable() {
+            isEnabled = true;
         }
     }
 
-    private enum ExceptionField {
-        CLASS("class"),
-        MESSAGE("message"),
-        STACKTRACE("stacktrace");
-
-        private final String val;
-
-        ExceptionField(String val) {
-            this.val = val;
+    private class RenderedFieldLabels {
+        final LoggerField locationClass;
+        final LoggerField locationFile;
+        final LoggerField locationLine;
+        final LoggerField locationMethod;
+
+        final LoggerField exceptionClass;
+        final LoggerField exceptionMessage;
+        final LoggerField exceptionStacktrace;
+
+        final LoggerField exception;
+        final LoggerField level;
+        final LoggerField location;
+        final LoggerField logger;
+        final LoggerField message;
+        final LoggerField mdc;
+        final LoggerField ndc;
+        final LoggerField host;
+        final LoggerField path;
+        final LoggerField tags;
+        final LoggerField timestamp;
+        final LoggerField thread;
+        final LoggerField version;
+
+        private final ArrayList<LoggerField> allFields = new ArrayList<LoggerField>();
+
+        RenderedFieldLabels() {
+            locationClass = new LoggerField("location.class", "class");
+            locationFile = new LoggerField("location.file", "file");
+            locationLine = new LoggerField("location.line", "line");
+            locationMethod = new LoggerField("location.method", "method");
+
+            exceptionClass = new LoggerField("exception.class", "class");
+            exceptionMessage = new LoggerField("exception.message", "message");
+            exceptionStacktrace = new LoggerField("exception.stacktrace", "stacktrace");
+
+            exception = new LoggerField("exception");
+            level = new LoggerField("level");
+            location = new LoggerField("location");
+            location.disable(); //By default location is not enabled as it's pretty expensive to resolve
+
+            logger = new LoggerField("logger");
+            message = new LoggerField("message");
+            mdc = new LoggerField("mdc");
+            ndc = new LoggerField("ndc");
+            host = new LoggerField("host");
+            path = new LoggerField("path");
+            tags = new LoggerField("tags");
+            timestamp = new LoggerField("@timestamp");
+            thread = new LoggerField("thread");
+            version = new LoggerField("@version");
+
+            List<LoggerField> locationFields = Arrays.asList(locationClass, locationFile, locationLine, locationMethod);
+            List<LoggerField> exceptionFields = Arrays.asList(exceptionClass, exceptionMessage, exceptionStacktrace);
+            List<LoggerField> rootFields = Arrays.asList(exception, level, location, logger, message, mdc, ndc, host, path, tags, timestamp, thread, version);
```

Comment:
It is possible to avoid the explicit list of all fields (it might be easy to forget one). For that, you may simply create a new function `loggerField` that created `LoggerField` with the constructor, adds the field into the list and returns the object back. Such side-effect may work nice, otherwise we may need to make sure all fields are included into the lists with tests